### ISSUE TITLE
readdir gives unexpected results

### ISSUE THREAD

**Issue by mickael-kerjean** [perry-mitchell/webdav-fs#56]
Let's say, my folder structure is the following:
```
test
  | test.txt
```

This code:
```
var wfs = require("webdav-fs")("http://127.0.0.1:8000");
wfs.readdir("/test/", function(err, c){
    console.log(c);
});
```
prints: ["test", "test.txt"] but I would have expect it to print ["test.txt"].

I've made up a repo on which you can test this: https://github.com/mickael-kerjean/webdav-fs:
```
npm install
node server_webdav.js &
node client_webdav.js
```
`[ 'level1', 'level2' ]`

The weird thing is it use to work correctly:
```
npm install --save webdav-fs@v1.5.0
node client_webdav.js
```
`[ 'level2' ]`

**Comment by perry-mitchell** [comment]
Thanks for posting @mickael-kerjean - It seems that this issue is reproducible in `webdav` as well, the root package. It looks that if I use `/level1` instead of `/level1/` (no trailing slash), it works correctly. Something to do with the filtering I guess.

**Comment by perry-mitchell** [comment]
Oh and thanks for the reproduction repo! It's a bit help.

**Comment by perry-mitchell** [comment]
Released in [v1.10.1](https://github.com/perry-mitchell/webdav-fs/blob/master/CHANGELOG.md#v1101). Thanks @mickael-kerjean
